src/diff/children.js

```diff
@@ -100,6 +100,11 @@ export function diffChildren(parentDom, newParentVNode, oldParentVNode, context,
 					// have a non-null _lastDomChild. Continue the diff from the end of
 					// this Fragment's DOM tree.
 					newDom = childVNode._lastDomChild;
+
+					// Eagerly cleanup _lastDomChild. We don't need to persist the value because
+					// it is only used by `diffChildren` to determine where to resume the diff after
+					// diffing Components and Fragments.
+					childVNode._lastDomChild = null;
```

Comment:
Instead of trying to keep `_lastDomChild` up to date, I figured it was better to just clean up the pointer after we no longer used it to reduce the likelihood of having out of date pointers to DOM nodes no longer in the DOM.